app/models/listed_taxon.rb

```diff
@@ -964,4 +965,35 @@ def establishment_means_description
     I18n.t( "establishment_means_descriptions.#{ key }", default: default )
   end
 
+  def reindex_observations_later
+    return true if taxon_id.blank? || place_id.blank?
+    return true unless previous_changes[:establishment_means]
+    ListedTaxon.delay(
+      priority: INTEGRITY_PRIORITY,
+      unique_hash: {
+        "CheckList::reindex_observations_later.taxon_id": taxon_id,
+        "CheckList::reindex_observations_later.place_id": place_id,
+      }
+    ).reindex_observations( taxon_id, place_id )
+    true
+  end
+
+  def self.reindex_observations( taxon_id, place_id )
+    page = 1
+    per_page = 1000
+    while true
+      res = Observation.elastic_search(
+        source: ["id"],
+        filters: [
+          { term: { private_place_ids: place_id } },
+          { term: { "taxon.ancestor_ids": taxon_id } }
+        ]
+      ).page( page ).per_page( per_page )
```

Comment:
Yep, this is the approach I was describing. Overall I think this will work much better. I did hit a problem here though with pagination of ES requests - if there are more than 10k taxa we'll hit the `index.max_result_window` error. We should use the `id_above` approach and filter by obs whose IDs are higher than the last ID of the previous batch.

I also noticed an existing small issue that we can address here or separately in the main brach. When given the delayed param, elastic_index! will still sleep in between queuing each batch to be indexed later when it doesn't need to - it just slows down creating all the sub-jobs. I think we should do something like `if batch_sleep && !options[:delay]` in acts_as_elastic_model.rb